### ISSUE TITLE
Is there a way to determine if a map entry is heap space with PVMMDLL_MAP_PTEENTRY?

### ISSUE THREAD

**Issue by rlewkowicz** [ufrisk/MemProcFS#54]
It's in fHeap for PVMMDLL_MAP_VADENTRY or you've got that VadMap_Type helper. Either that, or how can you determine the range between address with pVadMapEntry->vaStart and pVadMapEntry->vaEnd? I tried 

```
 pVadMapEntry->vaEnd -  pVadMapEntry->vaStart
```
But I ended up with a different result than (pPteMapEntry->cPages << 12) - 1

**Comment by ufrisk** [comment]
PTE map is not quite sufficient for that. PTE map is created from walking the page tables for PTE entries (and also potential non-valid software entries). It will condense consecutive PTEs with same flags into one entry.

A heap may be allocated in a VAD, but it may not be fully in-use yet (this will cause mismatch with # of PTE entries.
A heap may also have adjacent regions before and after it which may be clumped together in the same PTE entry, also causing mismatches.

In short, it's expected and normal to be discrepancies between PTE entries and heap region either from heap map or VAD map. If you check out the VadEx map or the files in the memmap\vad-v folder in the file system this will be a bit clearer.
